source/Framework Basics.rst

```diff
@@ -0,0 +1,173 @@
+
+    * ``interventionName``: This is the unique name of the intervention.
+    * ``shouldActivateFunc``: This function which tells whether this intervention should be activated or not. The function should return a Boolean value.
+    * ``shouldDeactivateFunc``: This function which tells whether this intervention should be deactivated. The function should return a Boolean value.
+    * ``firstTimeActionFunc``: This is an optional function which will be executed at the start of the intervention
+    * ``whenActiveActionFunc``: This is an optional function which will be executed per tick when intervention is active.
+
+Example using the ``Intervention`` object:
+````````````````````````````````````````````````````
+.. code-block:: scala
+
+    //This implements a lockdown when the nuber of agents in the infected compartment is greater than or equal to 2000
+    private def addLockdown(implicit context: Context): Unit = {
+
+    var interventionActivatedAt = 0
+    val interventionName = "lockdown"
+    val activationCondition = (context: Context) => getInfectedCount(context) >= 2000
+    val firstTimeExecution = (context: Context) => interventionActivatedAt = context.getCurrentStep
+    val deActivationCondition = (context: Context) => {
+    context.getCurrentStep >= interventionActivatedAt + 2*14
+    }
+    val intervention =
+    Intervention(interventionName, activationCondition, deActivationCondition, firstTimeExecution)
+
+    val lockdownSchedule = (myDay, myTick).add[House](0, 1)
+
+    registerIntervention(intervention)
+    registerSchedules(
+    (
+        lockdownSchedule,
+        (agent: Agent, context: Context) => {
+        val isEssentialWorker = agent.asInstanceOf[Person].isEssentialWorker
+        val isLockdown = context.activeInterventionNames.contains(interventionName)
+        isLockdown && !isEssentialWorker// && agent.asInstanceOf[Person].isInfected
+        },
+        1
+    )
+    )
+    }
+
+
+2.IntervalBasedIntervention
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This can be used to invoke an intervention which starts and end at specific ticks. This can be used for giving relaxations in the epidemic regulations for a specified period of time, for example during a festival.
+
+There are 5 paramertes that can be passed to the ``IntervalBasedIntervention`` object:
+
+    * ``interventionName``: This is the unique intervention name.
+    * ``startTick``: This ia an integer that specifies the start tick for intervention (inclusive); it should not be greater than endTick.
+    * ``endTick``: This is an integer that specifies the end tick for the intervention (exclusive, intervention will not be active at "endTick".)
+    * ``firstTimeActionFunc``: This is an optional function which gets executed when simulation starts.
+    * ``whenActiveActionFunc``: This is an optional function which gets executed per tick when simulation is active.
+
+Example suing the ``IntervalBasedIntervention`` object:
```

Comment:
Be careful with your spelling, you've made this mistake everywhere below as well.